relayer/src/foreign_client.rs

```diff
@@ -400,24 +400,27 @@ impl ForeignClient {
         &self,
         target_height: Height,
     ) -> Result<Vec<Any>, ForeignClientError> {
-        self.build_update_client_with_trusted(target_height, Height::zero())
+        self.build_update_client_with_trusted(target_height, None)
     }
 
     /// Returns a vector with a message for updating the client to height `target_height`.
     /// If the client already stores consensus states for this height, returns an empty vector.
     pub fn build_update_client_with_trusted(
         &self,
         target_height: Height,
-        trusted_height: Height,
+        trusted_height: Option<Height>,
     ) -> Result<Vec<Any>, ForeignClientError> {
+        let latest_height = || {
+            self.src_chain().query_latest_height().map_err(|e| {
+                ForeignClientError::ClientUpdate(format!(
+                    "failed fetching src chain latest height with error: {}",
+                    e
+                ))
+            })
+        };
+
         // Wait for source chain to reach `target_height`
-        while self.src_chain().query_latest_height().map_err(|e| {
-            ForeignClientError::ClientUpdate(format!(
-                "failed fetching src chain latest height with error: {}",
-                e
-            ))
-        })? < target_height
-        {
+        while latest_height()? < target_height {
```

Comment:
@ancazamfir Do we really want to wait here in a loop? Shouldn't this be handled via retries at the worker level?